tests/package/test_packagers_manager.py

```diff
@@ -0,0 +1,364 @@
+# Copyright 2018 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import os
+import shutil
+import tempfile
+import zipfile
+from typing import Any, Dict, List, Tuple, Type, Union
+
+import pytest
+
+from mlrun import DataItem
+from mlrun.artifacts import Artifact
+from mlrun.errors import MLRunInvalidArgumentError
+from mlrun.package import (
+    DefaultPackager,
+    MLRunPackagePackagerCollectionError,
+    Packager,
+    PackagersManager,
+)
+
+
+class PackagerA(Packager):
+    """
+    A simple packager to pack strings as results.
+    """
+
+    PACKABLE_OBJECT_TYPE = str
+
+    @classmethod
+    def get_default_artifact_type(cls, obj: Any) -> str:
+        return "result"
+
+    @classmethod
+    def get_supported_artifact_types(cls) -> List[str]:
+        return ["result"]
+
+    @classmethod
+    def is_packable(cls, object_type: Type, artifact_type: str = None) -> bool:
+        return object_type is cls.PACKABLE_OBJECT_TYPE and artifact_type == "result"
```

Comment:
Can't this (or a similar) implementation be included in the base class (`Packager`)? I would expect the default implementation to be:

```return object_type is cls.PACKABLE_OBJECT_TYPE and artifact_type in cls.get_supported_artifact_types()```

It doesn't support `...` as supported types, but we don't expect most packagers to support it, and if you want to support it then override the implementation (like the default-packager does). Having this in `Packager` will remove the need to implement it per packager.

Reply:
added :)